mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -2881,15 +2977,18 @@ static ModuleOp generateKernel(MLIRContext *context, GenParams &genParams,
     genParams.features = enabledFeatures;
     genParams.arch = arch;
     genParams.perfConfig = perfConfig;
-    if (isGemm) {
+    if (isGemm || isAttention) {
       for (const auto &arg :
            {filterDataType.getValue(), inputDataType.getValue(),
             outputDataType.getValue()})
         genParams.types.push_back(typeFromString(arg, context));
       genParams.convConfig = std::nullopt;
-      (void)createGpuGemmKernel(module, genParams);
+      if (isGemm) {
+        (void)createGpuGemmKernel(module, genParams);
+      } else {
+        (void)createGpuAttentionKernel(module, genParams);
```

Comment:
this is not related to a gemm; so might as well as a seperate stub for it.
Im a bit confused why/how filterDataType inputDataType outputDataType are relevant/working here.

Reply:
`filter`, `input`, and `output` are also `a`, `b`, and `c` - they're aliased

Reply:
I did something like this

https://github.com/ROCmSoftwarePlatform/rocMLIR/blob/754db02dfa0f55f6d0dc045513c40a0a1af3c9f3/mlir/tools/rocmlir-gen/rocmlir-gen.cpp#L2984-L2990